packages/cyberstorm/src/components/CustomSelect.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { ReactNode } from "react";
+import styles from "./componentStyles/CustomSelect.module.css";
+import { Button } from "./Button";
+import { CustomSelectItem, SelectItemProps } from "./CustomSelectItem";
+import { faChevronDown } from "@fortawesome/free-solid-svg-icons";
+import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
+
+import * as Select from "@radix-ui/react-select";
+import { SelectItem } from "@radix-ui/react-select";
+
+export type SelectOption = {
+  value: string;
+  content: typeof CustomSelectItem;
+  displayValue?: string;
+};
+
+type SelectProps = {
+  colorScheme?: "default" | "defaultDark" | "primary";
+  defaultOpen?: boolean;
+  icon?: ReactNode;
+  options?: SelectOption[];
+  setValue: CallableFunction;
+  value: string;
+};
+
+export const CustomSelect: React.FC<SelectProps> = (props) => {
+  const { colorScheme, defaultOpen, icon, options, setValue, value } = props;
+
+  const selectItemElements = options
+    ? mapSelectData(options, colorScheme)
+    : null;
+
+  return (
+    <div className={styles.root}>
+      <Select.Root
+        defaultOpen={defaultOpen}
+        value={value}
+        onValueChange={(newValue) =>
+          setValue !== undefined ? setValue(newValue) : null
+        }
+      >
+        <Select.Trigger asChild>
+          <div>
+            <Button
+              colorScheme={colorScheme}
+              rightIcon={icon}
+              label={
+                options?.find((o) => o.value === value)?.displayValue ?? ""
```

Comment:
The default value could be "Choose" oslt 🤷🏻 ?

Reply:
Thought about this and didn't come up with anything so I left it as blank, but "Choose" sounds good.

Reply:
How about `Select`? :)

Reply:
Leaving unresolved due to gkurck's comment. The default value is currently "Choose" as I suggested.

Reply:
It is now `Select` as suggested.